docs/source/user/saml_login.rst

```diff
@@ -0,0 +1,98 @@
+=====================
+Single Sign-On (SAML)
+=====================
+
+SAML login is supported by generating an additional temporary token right
+after users authenticates via SSO, the user is then redirected to the
+captive page with two querystring parameters: ``username`` and ``token``.
+
+The captive page must recognize these two parameters and automatically perform
+the submit action of the login form: ``username`` should obviously used for the
+username field, while ``token`` should be used for the password field.
+
+The internal REST API of openwisp-radius will recognize the token and authorize
+the user.
+
+This kind of implementation allows to implement the SAML login with any captive
+portal which already supports the RADIUS protocol because it's totally transparent
+for it, that is, the captive portal doesn't even know the user is signing-in with
+a SSO.
+
+.. note::
+   If you're building a public wifi service, we suggest
+   to take a look at `openwisp-wifi-login-pages <https://github.com/openwisp/openwisp-wifi-login-pages>`_,
+   which is built to work with openwisp-radius.
+
+Setup
+-----
+
+Install required dependencies::
+
+    pip install openwisp-radius[saml]
+
+Ensure your ``settings.py`` looks like the following:
+
+.. code-block:: python
+
+    INSTALLED_APPS = [
+        # ... other apps ..
+        # apps needed for SA<L> login
+        'rest_framework.authtoken',
+        'django.contrib.sites',
+        'allauth',
+        'allauth.account',
+        'djangosaml2'
+    ]
+
+    SITE_ID = 1
+
+    # Update AUTHENTICATION_BACKENDS
+    AUTHENTICATION_BACKENDS = (
+        'openwisp_users.backends.UsersAuthenticationBackend',
+        'djangosaml2.backends.Saml2Backend', # <- add for SAML login
+    )
+
+    # Update MIDDLEWARE
+    MIDDLEWARE = [
+        # ... other middlewares ...
+        'djangosaml2.middleware.SamlSessionMiddleware',
+    ]
+
+Ensure your main ``urls.py`` contains the ``allauth.urls``:
+
+.. code-block:: python
+
+    urlpatterns = [
+        # .. other urls ...
+        path('accounts/', include('allauth.urls')),
```

Comment:
why is this needed for SAML?

We normally have this:

```
url(r'^accounts/', include('openwisp_users.accounts.urls'))
```

Which includes the subset of URLs allauth.account URLs that are strictly needed for the service to work.
See https://github.com/openwisp/openwisp-users/blob/70e0ffa2522132a2cd3e277a447039e537231f95/openwisp_users/accounts/urls.py.

These URLs are automatically included by openwisp-controller: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/urls.py#L5-L10.

Reply:
I took reference from https://openwisp-radius.readthedocs.io/en/latest/user/social_login.html#setup.
Without accounts URL the SAML feature doesn't work. I have updated both occurrences

Reply:
Yeah I think that doc page is outdated too.
It should work with the url routes defined in openwisp-users, I tested this, please double check.